azurerm/internal/services/appplatform/resource_arm_spring_cloud_service.go

```diff
@@ -0,0 +1,703 @@
+												},
+
+												"host_key_algorithm": {
+													Type:     schema.TypeString,
+													Optional: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														"ssh-dss",
+														"ssh-rsa",
+														"ecdsa-sha2-nistp256",
+														"ecdsa-sha2-nistp384",
+														"ecdsa-sha2-nistp521",
+													}, false),
+												},
+
+												"strict_host_key_checking_enabled": {
+													Type:     schema.TypeBool,
+													Optional: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmSpringCloudServiceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppPlatform.ServicesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	existing, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for present of existing Spring Cloud %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_spring_cloud_service", *existing.ID)
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	resource := appplatform.ServiceResource{
+		Location: utils.String(location),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	// current create api doesn't take care parameters of config server.
+	// so we need to invoke create api first and then update api
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, resource)
+	if err != nil {
+		return fmt.Errorf("creating Spring Cloud %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation of Spring Cloud %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	gitPropertyRaw := d.Get("config_server_git_settings").([]interface{})
+	resource.Properties = &appplatform.ClusterResourceProperties{
+		ConfigServerProperties: &appplatform.ConfigServerProperties{
+			ConfigServer: &appplatform.ConfigServerSettings{
+				GitProperty: expandArmSpringCloudConfigServerGitProperty(gitPropertyRaw),
+			},
+		},
+	}
+
+	updateFuture, err := client.Update(ctx, resourceGroup, name, resource)
+	if err != nil {
+		return fmt.Errorf("failure updating config server of Spring Cloud Service %q  (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = updateFuture.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("failure waiting for setting config server of Spring Cloud Service %q config server (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
```

Comment:
I am sorry I don't understand what you mean.
Codes here is create first and update config server parameters. 
Because create rest api can only set up a spring cloud service cluster, It will not take care of config server parameters

Reply:
Hi @njuCZ 
I mean that if the user config does not specify a config repository at all, as you have done in your `_basic` test, we should probably avoid sending an additional API that just contains a `nil` for that property? (I'm not sure how much additional time that takes, but I know `Create` can take quite some time to execute).  
I'll review your latest changes now, we've had an internal discussion on the schema that I'll explain in the next review comments.

Reply:
done